GOFevaluation/utils.py

```diff
@@ -136,7 +135,29 @@ def _get_count_density(ns, be_first, be_second, data_sample):
     return ns
 
 
-def _get_equiprobable_binning(reference_sample, n_partitions, order=None):
+def _weighted_equi(n_bins, data, weights):
+    """Perform a 1D equiprobable binning for the reference sample. The binning
+    is defined such that the number of counts in each bin are (almost) equal.
+    Bins are defined based on the ECDF of the reference sample.
+    :param n_bins: number of partitions in this dimension
+    :type n_bins: int
+    :param data: sample of unbinned reference
+    :type data: array_like, 1-Dimensional
+    :param weights: weights of samples
+    :type weights: array_like, 1-Dimensional
```

Comment:
mention that it should have the same length as the data (reference) sample. Also, I think you should make an assertion that this is given. Right now it doesn't even complain when I pass a string 😄